### PR TITLE
Dispose created X509 certificates

### PR DESCRIPTION
**Pull request by tmahmood-microsoft**
https://github.com/Azure/azure-iot-sdk-csharp/blob/672b447887df5faf0018d279d4508462a761c388/iothub/device/src/Edge/TrustBundleProvider.cs#L84

These created X509Certificate2 instances don't appear to be disposed anywhere, but they need to be.

### CODE REVIEW

File: iothub/device/src/Edge/InstalledCertificateValidator.cs [hunk 1]
```diff
@@ -48,4 +57,4 @@ private void SetupCertificateValidation()
             }
         }
     }
-}
```
Comment (patilsnr) [inline iothub/device/src/Edge/InstalledCertificateValidator.cs:51]: nit: newline at end of file

**Review by patilsnr — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

File: iothub/device/src/Edge/CustomCertificateValidator.cs [hunk 2]
```diff
@@ -10,11 +10,20 @@
...
 namespace Microsoft.Azure.Devices.Client
 {
-    internal class CustomCertificateValidator : ICertificateValidator
+    internal class CustomCertificateValidator : ICertificateValidator, IDisposable
     {
         private readonly IEnumerable<X509Certificate2> _certs;
         private readonly IotHubClientTransportSettings _transportSettings;
 
+        /// <inheritdoc/>
+        public void Dispose()
```
Comment (drwill-ms) [inline iothub/device/src/Edge/CustomCertificateValidator.cs:19]: methods below ctors, even if they are public and the ctor isn't.

File: iothub/device/src/Edge/ICertificateValidator.cs [hunk 3]
```diff
@@ -7,7 +7,7 @@
 
 namespace Microsoft.Azure.Devices.Client
 {
-    internal interface ICertificateValidator
+    internal interface ICertificateValidator : IDisposable
```
Comment (drwill-ms) [inline iothub/device/src/Edge/ICertificateValidator.cs:10]: No need to make this IDisposable too.

File: iothub/device/src/IotHubModuleClient.cs [hunk 4]
```diff
@@ -280,6 +280,7 @@ private async Task<DirectMethodResponse> InvokeMethodAsync(Uri uri, DirectMethod
             finally
             {
                 httpClientHandler?.Dispose();
+                _certValidator.Dispose();
```
Comment (drwill-ms) [inline iothub/device/src/IotHubModuleClient.cs:283]: Does this need NRE protection?

**Review by drwill-ms — APPROVED** [review]

File: iothub/device/src/IotHubModuleClient.cs (hunk 4 above)
Comment (tmahmood-microsoft) [reply iothub/device/src/IotHubModuleClient.cs:283]: It does not. We create an instance NullCertificateValidator in such case.

**Pull request merged**